File: Scripts/results/generated_code_results/baseline_claude/Education/OnlineTeachingSoftware/DiscoverScreen.js
import React from 'react';

import { View, Text, Image, StyleSheet, TouchableOpacity, ScrollView, SafeAreaView } from 'react-native';

import { useNavigation } from '@react-navigation/native';

import Icon from 'react-native-vector-icons/Ionicons';



const DiscoverScreen = () => {

  const navigation = useNavigation();



  const renderPopularCourse = (title, instructor, price, image) => (

    <TouchableOpacity style={styles.popularCourseItem} onPress={() => navigation.navigate('CourseDetail')}>

      <Image source={image} style={styles.popularCourseImage} />

      <View style={styles.popularCourseInfo}>

        <Text style={styles.popularCourseCategory}>{title.split(' ')[0]}</Text>

        <Text style={styles.popularCourseTitle}>{title}</Text>

        <Text style={styles.popularCourseInstructor}>{instructor}</Text>

        <Text style={styles.popularCoursePrice}>${price}</Text>

      </View>

      <TouchableOpacity style={styles.favoriteButton}>

        <Icon name="heart-outline" size={20} color="#000" />

      </TouchableOpacity>

    </TouchableOpacity>

  );



  return (

    <SafeAreaView style={styles.container}>

      <ScrollView>

        <View style={styles.header}>

          <TouchableOpacity>

            <Icon name="menu-outline" size={24} color="#fff" />

          </TouchableOpacity>

          <Text style={styles.headerTitle}>Discover</Text>

          <TouchableOpacity>

            <Icon name="notifications-outline" size={24} color="#fff" />

          </TouchableOpacity>

        </View>



        <View style={styles.heroSection}>

          <Text style={styles.heroTitle}>Study what you want</Text>

          <Text style={styles.heroSubtitle}>anytime anywhere</Text>

        </View>



        <View style={styles.featuredCourseCard}>

          <View style={styles.featuredCourseInfo}>

            <Text style={styles.featuredCourseCategory}>Interface Design</Text>

            <Text style={styles.featuredCourseTitle}>User interface for beginners</Text>

            <View style={styles.ratingContainer}>

              <Icon name="star" size={16} color="#FFD700" />

              <Text style={styles.rating}>4.7 (1,242 Reviews)</Text>

            </View>

            <View style={styles.progressContainer}>

              <View style={[styles.progressBar, { width: '48%' }]} />

            </View>

            <Text style={styles.progressText}>48%</Text>

          </View>

          <Image source={require('../assets/snack-icon.png')} style={styles.featuredCourseImage} />

        </View>



        <View style={styles.popularCoursesSection}>

          <View style={styles.sectionHeader}>

            <Text style={styles.sectionTitle}>Popular Course</Text>

            <TouchableOpacity>

              <Text style={styles.seeAllButton}>See All</Text>

            </TouchableOpacity>

          </View>

          <ScrollView horizontal showsHorizontalScrollIndicator={false}>

            {renderPopularCourse('Expert Wireframing for Mobile Design', 'Jerremy Mamika', '48', require('../assets/snack-icon.png'))}

            {renderPopularCourse('Bootstrapping Principles', 'Alexander Be', '48', require('../assets/snack-icon.png'))}

          </ScrollView>

        </View>

      </ScrollView>



      <View style={styles.tabBar}>

        <TouchableOpacity style={styles.tabItem}>

          <Icon name="home" size={24} color="#6C5CE7" />

          <Text style={[styles.tabLabel, styles.activeTabLabel]}>Home</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.tabItem} onPress={() => navigation.navigate('Statistics')}>

          <Icon name="stats-chart-outline" size={24} color="#999" />

          <Text style={styles.tabLabel}>Statistic</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.tabItem} onPress={() => navigation.navigate('Favorites')}>

          <Icon name="heart-outline" size={24} color="#999" />

          <Text style={styles.tabLabel}>Favorites</Text>

        </TouchableOpacity>

        <TouchableOpacity style={styles.tabItem}>

          <Icon name="person-outline" size={24} color="#999" />

          <Text style={styles.tabLabel}>Profile</Text>

        </TouchableOpacity>

      </View>

    </SafeAreaView>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#1E1E2E',

  },

  header: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    padding: 16,

  },

  headerTitle: {

    fontSize: 20,

    fontWeight: 'bold',

    color: '#fff',

  },

  heroSection: {

    padding: 16,

  },

  heroTitle: {

    fontSize: 24,

    fontWeight: 'bold',

    color: '#fff',

  },

  heroSubtitle: {

    fontSize: 24,

    fontWeight: 'bold',

    color: '#fff',

    marginTop: 4,

  },

  featuredCourseCard: {

    backgroundColor: '#fff',

    borderRadius: 12,

    margin: 16,

    padding: 16,

    flexDirection: 'row',

  },

  featuredCourseInfo: {

    flex: 1,

  },

  featuredCourseCategory: {

    color: '#6C5CE7',

    fontWeight: '500',

    marginBottom: 4,

  },

  featuredCourseTitle: {

    fontSize: 18,

    fontWeight: 'bold',

    marginBottom: 8,

  },

  ratingContainer: {

    flexDirection: 'row',

    alignItems: 'center',

    marginBottom: 8,

  },

  rating: {

    marginLeft: 4,

  },

  progressContainer: {

    height: 4,

    backgroundColor: '#E0E0E0',

    borderRadius: 2,

    marginBottom: 4,

  },

  progressBar: {

    height: 4,

    backgroundColor: '#6C5CE7',

    borderRadius: 2,

  },

  progressText: {

    alignSelf: 'flex-end',

  },

  featuredCourseImage: {

    width: 100,

    height: 100,

    borderRadius: 8,

  },

  popularCoursesSection: {

    marginTop: 16,

  },

  sectionHeader: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    paddingHorizontal: 16,

    marginBottom: 12,

  },

  sectionTitle: {

    fontSize: 18,

    fontWeight: 'bold',

    color: '#fff',

  },

  seeAllButton: {

    color: '#6C5CE7',

  },

  popularCourseItem: {

    width: 240,

    backgroundColor: '#fff',

    borderRadius: 12,

    marginLeft: 16,

    marginBottom: 16,

    overflow: 'hidden',

  },

  popularCourseImage: {

    width: '100%',

    height: 120,

  },

  popularCourseInfo: {

    padding: 12,

  },

  popularCourseCategory: {

    color: '#6C5CE7',

    marginBottom: 4,

  },

  popularCourseTitle: {

    fontSize: 16,

    fontWeight: 'bold',

    marginBottom: 4,

  },

  popularCourseInstructor: {

    color: '#666',

    marginBottom: 4,

  },

  popularCoursePrice: {

    fontWeight: 'bold',

  },

  favoriteButton: {

    position: 'absolute',

    top: 8,

    right: 8,

    backgroundColor: '#fff',

    borderRadius: 16,

    padding: 4,

  },

  tabBar: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    borderTopWidth: 1,

    borderTopColor: '#333',

    paddingVertical: 8,

    backgroundColor: '#1E1E2E',

  },

  tabItem: {

    alignItems: 'center',

  },

  tabLabel: {

    marginTop: 4,

    fontSize: 12,

    color: '#999',

  },

  activeTabLabel: {

    color: '#6C5CE7',

  },

});



export default DiscoverScreen;

File: Scripts/results/generated_code_results/Flutter/bak_code/Education/OnlineTeachingSoftware/DiscoverScreen.dart
import 'package:flutter/material.dart';
class DiscoverScreen extends StatefulWidget {
@override
_DiscoverScreenState createState() => _DiscoverScreenState();
}
class _DiscoverScreenState extends State<DiscoverScreen> {
int _selectedIndex = 0;
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Discover'),
leading: IconButton(
icon: Icon(Icons.grid_view),
onPressed: () {},
),
actions: [
IconButton(
icon: Stack(
children: [
Icon(Icons.notifications),
Positioned(
right: 0,
top: 0,
child: Container(
padding: EdgeInsets.all(2),
decoration: BoxDecoration(
color: Colors.red,
borderRadius: BorderRadius.circular(6),
),
constraints: BoxConstraints(
minWidth: 12,
minHeight: 12,
),
child: Text(
'',
style: TextStyle(
color: Colors.white,
fontSize: 8,
),
textAlign: TextAlign.center,
),
),
)
],
),
onPressed: () => Navigator.pushNamed(context, 'Notifications'),
),
],
),
body: SingleChildScrollView(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Padding(
padding: const EdgeInsets.all(16.0),
child: Text(
'Study what you want\nanytime anywhere',
style: Theme.of(context).textTheme.headline5,
),
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Card(
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text('Interface Design'),
SizedBox(height: 8),
Text(
'User interface for beginners',
style: Theme.of(context).textTheme.headline6,
),
SizedBox(height: 8),
Row(
children: [
Icon(Icons.star, color: Colors.yellow),
Text('4.7 (1,242 Reviews)'),
],
),
SizedBox(height: 8),
LinearProgressIndicator(value: 0.48),
SizedBox(height: 8),
Text('48% completed'),
SizedBox(height: 8),
Image.asset('assets/images/template.png'),
],
),
),
),
),
Padding(
padding: const EdgeInsets.symmetric(horizontal: 16.0),
child: Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text(
'Popular Course',
style: Theme.of(context).textTheme.headline6,
),
TextButton(
onPressed: () {},
child: Text('See All'),
),
],
),
),
ListView(
shrinkWrap: true,
physics: NeverScrollableScrollPhysics(),
children: [
_buildCourseCard(
'Expert Wireframing for Mobile Design',
'Graphic Design',
'Jerremy Mamika',
'$48',
),
_buildCourseCard(
'Bootstrapping Principles',
'Development',
'Alexander Be',
'',
),
],
),
],
),
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: _selectedIndex,
onTap: (index) {
setState(() {
_selectedIndex = index;
});
switch (index) {
case 0:
break;
case 1:
Navigator.pushNamed(context, 'Statistics');
break;
case 2:
Navigator.pushNamed(context, 'Favorites');
break;
case 3:
break;
}
},
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.bar_chart), label: 'Statistic'),
BottomNavigationBarItem(icon: Icon(Icons.favorite), label: 'Favorites'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),
],
),
);
}
Widget _buildCourseCard(String title, String category, String instructor, String price) {
return Card(
margin: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
child: InkWell(
onTap: () => Navigator.pushNamed(context, 'CourseDetail'),
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Row(
children: [
Expanded(
flex: 2,
child: Image.asset('assets/images/template.png'),
),
SizedBox(width: 16),
Expanded(
flex: 3,
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(category, style: TextStyle(color: Colors.purple)),
SizedBox(height: 4),
Text(title, style: Theme.of(context).textTheme.subtitle1),
SizedBox(height: 4),
Text(instructor, style: TextStyle(color: Colors.grey)),
if (price.isNotEmpty) SizedBox(height: 4),
if (price.isNotEmpty) Text(price, style: TextStyle(fontWeight: FontWeight.bold)),
],
),
),
IconButton(
icon: Icon(Icons.favorite_border),
onPressed: () {},
),
],
),
),
),
);
}
}